launcher/src/main/java/org/apache/spark/launcher/Main.java

```diff
@@ -101,6 +110,80 @@ public static void main(String[] argsArray) throws Exception {
     }
   }
 
+  private static void printUsage() {
```

Comment:
Agree with Sean this copy-paste really not a good way to handle this issue. we should find out the root cause of this broken compared to 1.6 and then try to fix in that way.